lib/http.dart

```diff
@@ -125,8 +125,10 @@ Future<Response> patch(url,
 /// the same server, you should use a single [Client] for all of those requests.
 ///
 /// For more fine-grained control over the request, use [Request] instead.
-Future<Response> delete(url, {Map<String, String> headers}) =>
-    _withClient((client) => client.delete(url, headers: headers));
+Future<Response> delete(url,
+        {Map<String, String> headers, body, Encoding encoding}) =>
```

Comment:
Could you tell me please, what is the exact type definition for the body?

Reply:
I can directly send a Map to body, which tries to convert it to (JSON) String, or String. The body field work the same across HTTP verb methods, where it's not included in the GET or OPTION as these verbs doesn't support request body and inside it. Meanwhile, DELETE does support body, hence this pull request. 

You can observe that type definition for body isn't also referenced in other methods in this file. Hence my support for this PR

Reply:
We lack documentation of the `body` parameter on the other similar methods too, and the documentation in general isn't amazing, so we should give it an overhaul soon.
No need to do anything *special* here, it's the same `body` as all the other `body` parameters.

Reply:
Do you have plans regarding the documentation overhaul? If you share your plans the community might be able to help.